e2e/deployers/retry.go

```diff
@@ -46,3 +47,31 @@ func waitSubscriptionPhase(namespace, name string, phase subscriptionv1.Subscrip
 		time.Sleep(time.Second * time.Duration(util.TimeInterval))
 	}
 }
+
+func waitDeploymentReady(client client.Client, namespace, name string) error {
```

Comment:
This should change to workload.Health, where the workload know what artifact(s) it should check and report health back to the deployer or where needed.

Reply:
In my mind, In theory, workload should provide resource list, and deployer should check workload health. This is because only deployer know where the workload is deployed to (name/namespace/cluster).

But with subscription and appset, the resource list are from an online git url and also we add local kustomization to it. How could we get the resource list from our created subscription and appset CR?

Reply:
> In my mind, In theory, workload should provide resource list, and deployer should check workload health. This is because only deployer know where the workload is deployed to (name/namespace/cluster).

If we did this, the deployer would need to change for each workload type. For example in the future as we handle VitualMachine resource it would need the deployer to know about such resources.

The workload on the other hand knows about its resources and can check its health, given where it is deployed (IOW, cluster, namespace, name, etc.)

> 
> But with subscription and appset, the resource list are from an online git url and also we add local kustomization to it. How could we get the resource list from our created subscription and appset CR?

We do not need to. The Workload is tailored to a specfic workload in the samples, so the workload knows what it is deploying etc. and can leverage the same scheme as discovered when it is passed the required manner of identifying where it is deployed.